mount/linux-lib.pl

```diff
@@ -905,8 +905,8 @@ sub disk_space
 &clean_language();
 local $out = &backquote_command("df -k ".quotemeta($_[1]), 1);
 &reset_environment();
-if ($out =~ /Mounted on\n\S+\s+(\S+)\s+\S+\s+(\S+)/) {
-	return ($1, $2);
+if ($out =~ /Mounted on\n\S+\s+(?<total>\S+)\s+(?<used>\S+)\s+(?<free>\S+)\s+(?<percent>\d+)/) {
+	return ("$+{total}", "$+{free}", "$+{used}", "$+{percent}");
```

Comment:
Can we not just compute the percent rather than fetching it from the `df` command?

Reply:
We are only fetching it from `df`. And compute if unavailable.

Reply:
But why not make this regexp simpler and just always compute it?

Reply:
Oh, I see. Well, we cannot change the returned params order! This is made not to always compute percent or not. This is made to make sure that we always first return `total` and `free`, and a new additional params `used` and `percent` .. this is what caller expects. There are other non Linux libs and they only (for now) return old type of data (i.e. `total` and `free`).

In other words, we need to preserve the return order of the params but it's impossible to do using regex in any other way because you were initially skipping on `used` with this old regex `\S+\s+(\S+)\s+\S+\s+(\S+)/`..

Reply:
Oh I see ... in that case, let's keep it the way it is then. I assume that all Linux versions use the same format for the `df` command.

Reply:
Alright, fixed on the last commit.